app/Helpers/GeometryHelper.php

```diff
@@ -117,4 +119,29 @@ public static function getPolygonsBbox($polygonsIds)
 
         return [$minX, $minY, $maxX, $maxY];
     }
+
+    public static function getCriteriaDataForPolygonGeometry($polygonGeometry)
+    {
+        // Fetch data from criteria_site with distinct criteria_id based on the latest created_at
+        $criteriaDataQuery = 'SELECT criteria_id, MAX(created_at) AS latest_created_at
+                            FROM criteria_site 
+                            WHERE polygon_id = ?
+                            GROUP BY criteria_id';
+
+        $criteriaData = DB::select($criteriaDataQuery, [$polygonGeometry->uuid]);
```

Comment:
Please lean into the Laravel associations as much as possible. For one thing, this query as written is completely ignoring the soft deletes feature. This should get you what you want instead:
```
$polygonGeometry->criteriaSite()->selectRaw('criteria_id, max(created_at) as latest_created_at')->groupBy('criteria_id')->get();
```

Reply:
really great improvement.
implemented